### PR TITLE
Cht tests fazzdev #1321 33

### PR DESCRIPTION
**Pull request by thefazzer**
**Initial check in Issue 33 (#1321)**

- First header in the message is HEADERS_START and last is HEADERS_END).
- AV = assume valid header, CP1,CP2 - checkpointed headers. '---' some headers.
- HEADERS_START--CP1----AV----HEADERS_END---CP2---.
- Check that headers till AV(including AV) are marked for download.
- Headers after AV are not marked for download.

Attention: @Aprogiena @noescape00

### CODE REVIEW

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs [hunk 1]
```diff
@@ -1284,5 +1284,51 @@ public void ChainWithMaxReorgPlusExtraHeadersIsCalled_AnotherChainIsPresented_Co
...
+            // Exclude second checkpoint at h45.
+            listOfChainHeaders = listOfChainHeaders.Take(40).ToList();
+            ConnectNewHeadersResult connectedHeadersResultNew = chainedHeaderTree.ConnectNewHeaders(1, listOfChainHeaders);
+
+            // From initialised chain up to and including assume valid (h6->h30) are marked for download.
+            // Headers after assume valid (h31->h40) are not marked for download.
+            IEnumerable<BlockHeader> extendedChain = listOfChainHeaders.Skip(initialChainSize);
+            
+            Assert.Equal(extendedChain.First(), connectedHeadersResultNew.DownloadFrom.Header);
+            Assert.Equal(assumeValidHeader.Header, connectedHeadersResultNew.DownloadTo.Header);
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: All good here, just please include checking of block data availability state for all these headers that are marked for download as well as for those that are not marked for download after assumed valid block.

**Comment by noescape00** [comment]
Conflicts

File: src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs (hunk 1 above)
Comment (monsieurleberre) [reply src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: Love your use of CTRL+K+D

Comment (thefazzer) [reply src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs]: Fixed!

**Comment by thefazzer** [comment]
Merge conflicts fixed.

**Pull request merged**